test/libsolidity/semanticTests/functionTypes/comparison_operators_for_external_functions.sol

```diff
@@ -0,0 +1,18 @@
+Contract C {
+    function () external externalStorage;
+
+    function comparePtr() public {
+        function () external externalLocal1;
+        function () external externalLocal2;
+
+        externalLocal1 == externalLocal2;
+        externalLocal1 != externalLocal2;
+
+        externalLocal1 == externalStorage;
+        externalStorage != externalLocal2;
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// comparePtr() ->
```

Comment:
Might need some adjustment, but maybe something along the lines of the following? Might be nicer to split into several cases, though.

```suggestion
contract C {
    function f() external {}
    function g() external {}

    function check() public returns(bool) {
        assert(f != g && f == f && g == g);
        function () external f_local = f;
        function () external g_local = g;
        assert(f != g_local && f_local != g && f_local == f && g_local == g);
        assert(f_local == f_local && f_local != g_local);
        assert(g_local != f_local && g_local == g_local);
        return true;
    }
}
// ====
// compileViaYul: also
// ----
// check() -> 1
```